src/systems/ec/implementations/vanilla/packages/ec-component-table/ec-component-table.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable unicorn/consistent-function-scoping */
+import { queryOne, queryAll } from '@ecl/ec-base/helpers/dom';
+
+/**
+ * @param {HTMLElement} element DOM element for component instantiation and scope
+ * @param {Object} options
+ * @param {String} options.sortSelector Selector for toggling element
+ * @param {Boolean} options.attachClickListener Whether or not to bind click events on toggle
+ */
+export class Table {
+  /**
+   * @static
+   * Shorthand for instance creation and initialisation.
+   *
+   * @param {HTMLElement} root DOM element for component instantiation and scope
+   *
+   * @return {Table} An instance of table.
+   */
+  static autoInit(root, { TABLE: defaultOptions = {} } = {}) {
+    const table = new Table(root, defaultOptions);
+    table.init();
+    // eslint-disable-next-line no-param-reassign
+    root.ECLTable = table;
+    return table;
+  }
+
+  constructor(
+    element,
+    {
+      sortSelector = '[data-ecl-tablesort-toggle]',
```

Comment:
I'm not sure `data-ecl-table`  https://github.com/ec-europa/europa-component-library/pull/1598/files#diff-6b789af6f8f8d406236aed4ca6aae98aR20 is necessary given this option is actually used for internal references within the class.

The `data-ecl-auto-init="Table"` referes to `root`/`element` which is enough I think.